harness_suite.go

```diff
@@ -0,0 +1,93 @@
+package poplar
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/mongodb/grip"
+)
+
+// BenchmarkSuite is a convenience wrapper around a group of suites.
+//
+// You can use the Standard() function to convert these benchmarks to
+// a standard go library benchmark function.
+type BenchmarkSuite []*BenchmarkCase
+
+// Validate aggregates the validation of all constituent tests. The
+// validation method for case may also modify a case to set better
+// defaults. Validate has continue-on-error semantics.
+func (s BenchmarkSuite) Validate() error {
+	catcher := grip.NewCatcher()
+
+	for _, c := range s {
+		catcher.Add(c.Validate())
+	}
+
+	return catcher.Resolve()
+}
+
+// Add creates and adds a new case to an existing suite and allows
+// access to a fluent-style API for declaring and modifying cases.
+func (s *BenchmarkSuite) Add() *BenchmarkCase {
```

Comment:
It's strange to have some methods on a pointer and others not, and can make interfaces hard.

Reply:
Go's dispatcher is smart enough to do the right thing *unless* you need to cast something to an interface that's implemented by both pointer and value receivers. As this isn't an interface, I think it's OK in the current form, particularly because the mutability implications given that BenchmarkSuite is a slice (and therefore already a pointer, in a way,) and the fact that most of the methods (except for this one) don't need to mutate the underlying structure.